arch/x86/deflate_quick.c

```diff
@@ -237,7 +243,7 @@ ZLIB_INTERNAL block_state deflate_quick(deflate_state *s, int flush) {
             hash_head = quick_insert_string(s, s->strstart);
             dist = s->strstart - hash_head;
 
-            if (dist > 0 && (dist-1) < (s->w_size - 1)) {
+            if (dist > 0 && (dist-1) < wsize) {
```

Comment:
Should be ```wsize - 1```...

Reply:
I think here it should be `wsize`. I don't see the need for `-1` .

Reply:
It's to avoid overflow... Obviously there has to be at least MIN_MATCH bytes before the match, otherwise there would not be anything to match with. So dist can't be more than ```wsize - MIN_MATCH```.

Reply:
I thought here that `dist` refers to the distance where the match is found in the window, not the distance + `MIN_MATCH`. If `MIN_MATCH` is 3 then why isn't it `w_size - 3`?

Reply:
In the worst case the original string starts at byte index 0, and match can be last 3 bytes of window (32765, 32766, 32767). I don't know why it isn't ```w_size - 3```, that is why I said the original test is also incorrect...

Reply:
Let's move the conversation to the other PR #541. Sorry for the confusion.